### DISCUSSION TITLE
[Bug]: Codeception - Module \Tests\Support\Helper\Functional could not be found and loaded

### DISCUSSION THREAD

**Discussion by pfilipkowskilemon** [pimcore/pimcore#16168] [Q&A / Support]
### Pimcore version

11.0.11

### Steps to reproduce

1. Add to composer.json "preferred-install": {
      "pimcore/pimcore": "source",
      "*": "dist"
    },
2. Delete the vendor directory, then download it again so that the composer can download the source version of Pimcore
3. Copy the _bootstrap.php and tests/functional.suite.yml configurations from the documentation
4. Try to run tests by vendor/bin/codecept run Functional

### Actual Behavior

Codeception throws error Module \Tests\Support\Helper\Functional could not be found and loaded

![Screenshot from 2023-10-20 10-05-14](https://github.com/pimcore/pimcore/assets/110021871/8e75a051-e3de-4ce1-aa8e-b34ccfbc8c7c)

### Expected Behavior

Codeception runs tests with Pimcore helpers with configuration taken from documentation.

**Comment by dvesh3** [comment]
This is not Pimcore issue. Seems like Functional Helper class is missing in your setup as defined in the config.

**Reply by pfilipkowskilemon** [reply]
I copied one-to-one the configuration from the documentation into Pimcore skieleton. Removing the line related to Functional Helper class Codeception continues to throw the same error.
![Screenshot from 2023-10-31 09-08-42](https://github.com/pimcore/pimcore/assets/110021871/7fadc2b9-bb73-48c0-baae-204f7ada48e4)

**Reply by VenzDev** [reply]
I'm struggling with exactly the same problem.

**Reply by william-siteone** [reply]
> This is not Pimcore issue. Seems like Functional Helper class is missing in your setup as defined in the config.

Well, yes it is.

When you pull the distribution of `pimcore/pimcore` via composer, the result is that the entire `vendor/pimcore/pimcore/tests/` folder is not part of the download. Which obviously leads to the problem of missing file `vendor/pimcore/pimcore/tests/Support/Util/Autoloader.php`.
  
Can somebody fix that? Can you please make the helpers available in the expected way via composer? CI/CD environments are having trouble "cloning" the pimcore/pimcore package instead of pulling like normal. 

To say it's not a Pimcore problem is very misleading.

**Official Documentation**
https://pimcore.com/docs/platform/Pimcore/Development_Tools_and_Details/Testing/Application_Testing/#codeception

> The tests/unit.suite.yml should be fine for a standard unit testing setup without dependencies, but we need to alter the functional test suite to initialize a test database and to boot Pimcore's kernel before running tests. Configure the suite to use the `\Pimcore\Tests\Support\Helper\Pimcore` helper:

How are we going to include the helper Pimcore tells us to include if it is not provided it via composer? Can someone please follow through on the whole issue instead of offering a knee-jerk response?

Thanks!

**Comment by patriciomg** [comment]
hi,
sadly I cannot find the version 11.0.11 to test it, it seems that is not available anymore.
I would rather, if possible, update pimcore to its last version, I have 11.1.4 and I haven't written any test yet, nor copied any file, i just got into the docker container and ran "vendor/bin/codecept run Functional" with no errors at all. Sorry i can't help further than this.
Greetings.
